layer/migration.go

```diff
@@ -178,6 +173,18 @@ func (ls *layerStore) RegisterByGraphID(graphID string, parent ChainID, tarDataF
 		referenceCount: 1,
 		layerStore:     ls,
 		references:     map[Layer]struct{}{},
+		diffID:         diffID,
+		size:           size,
+		chainID:        createChainIDFromParent(parent, diffID),
+	}
+
+	ls.layerL.Lock()
```

Comment:
This function is actually never called in parallel so these locks don't block in migration. Still, it should be correct of course. Copying/writing files when I already know that I won't be using them does not sound like a good idea, especially when this PR is trying to optimize things. I'm fine for the second lock for clarity.